src/derivative_operators/abstract_operator_functions.jl

```diff
@@ -0,0 +1,211 @@
+# ~ bound checking functions ~
+checkbounds(A::AbstractDerivativeOperator, k::Integer, j::Integer) =
+    (0 < k ≤ size(A, 1) && 0 < j ≤ size(A, 2) || throw(BoundsError(A, (k,j))))
+
+checkbounds(A::AbstractDerivativeOperator, kr::Range, j::Integer) =
+    (checkbounds(A, first(kr), j); checkbounds(A,  last(kr), j))
+
+checkbounds(A::AbstractDerivativeOperator, k::Integer, jr::Range) =
+    (checkbounds(A, k, first(jr)); checkbounds(A, k,  last(jr)))
+
+checkbounds(A::AbstractDerivativeOperator, kr::Range, jr::Range) =
+    (checkbounds(A, kr, first(jr)); checkbounds(A, kr,  last(jr)))
+
+checkbounds(A::AbstractDerivativeOperator, k::Colon, j::Integer) =
+    (0 < j ≤ size(A, 2) || throw(BoundsError(A, (size(A,1),j))))
+
+checkbounds(A::AbstractDerivativeOperator, k::Integer, j::Colon) =
+    (0 < k ≤ size(A, 1) || throw(BoundsError(A, (k,size(A,2)))))
+
+
+# BandedMatrix{A,B,C,D}(A::DerivativeOperator{A,B,C,D}) = BandedMatrix(full(A, A.stencil_length), A.stencil_length, div(A.stencil_length,2), div(A.stencil_length,2))
+
+# ~~ getindex ~~
+@inline function getindex(A::AbstractDerivativeOperator, i::Int, j::Int)
+    @boundscheck checkbounds(A, i, j)
+    mid = div(A.stencil_length, 2) + 1
+    bpc = A.stencil_length - mid
+    l = max(1, low(j, mid, bpc))
+    h = min(A.stencil_length, high(j, mid, bpc, A.stencil_length, A.dimension))
+    slen = h - l + 1
+    if abs(i - j) > div(slen, 2)
+        return 0
+    else
+        return A.stencil_coefs[mid + j - i]
+    end
+end
+
+# scalar - colon - colon
+@inline getindex(A::AbstractDerivativeOperator, kr::Colon, jr::Colon) = full(A)
+
+@inline function getindex(A::AbstractDerivativeOperator, rc::Colon, j)
+    T = eltype(A.stencil_coefs)
+    v = zeros(T, A.dimension)
+    v[j] = one(T)
+    copy!(v, A*v)
+    return v
+end
+
+
+# symmetric right now
+@inline function getindex(A::AbstractDerivativeOperator, i, cc::Colon)
+    T = eltype(A.stencil_coefs)
+    v = zeros(T, A.dimension)
+    v[i] = one(T)
+    copy!(v, A*v)
+    return v
+end
+
+
+# UnitRanges
+@inline function getindex(A::AbstractDerivativeOperator, rng::UnitRange{Int}, cc::Colon)
+    m = full(A)
```

Comment:
That looks like a memory usage surprise, but for now that's fine.

Reply:
Yeah since that's allocated, might as well get a view from it.

Reply:
Well actually, the type would be different than what one might expect... we can discuss that further in an issue though.